contract/src/lib.rs

```diff
@@ -540,6 +565,136 @@ impl BtcLightClient {
         );
     }
 
+    #[cfg(feature = "zcash")]
+    fn zcash_check_pow_and_equihash(
```

Comment:
I strongly suggest including comments referencing the corresponding zcashd code that this is based on. For example, lines 576-644 here correspond to [lines 32-54 of `GetNextWorkRequired`](https://github.com/zcash/zcash/blob/v6.2.0/src/pow.cpp#L32-L54) in zcashd 6.2.0. (It seems to have been ported from zcashd rather than Zebra, which is fine.)